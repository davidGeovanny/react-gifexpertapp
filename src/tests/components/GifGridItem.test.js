import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas de GifGridItem', () => {
    const title = 'Un t√≠tulo';
    const url = 'https://localhost/image.jpg';

    let wrapper = shallow( 
        <GifGridItem 
            title={ title } 
            url={ url } 
        /> 
    );
    
    test('Debe de renderizarse el componente y hacer match con el snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Should have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('Should have an image equal to url and alt of the props', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('should have a class called animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBe( true ); 
    });
    
});
