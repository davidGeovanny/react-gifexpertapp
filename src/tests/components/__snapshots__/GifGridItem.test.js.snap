// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem Debe de renderizarse el componente y hacer match con el snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/image.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
